packages/access/lib/grants.js

```diff
@@ -47,6 +50,25 @@ const evaluateConstraints = async (granter, campaign, email, t, pgdb) => {
   return { errors }
 }
 
+const insert = async (granter, campaignId, grants = [], pgdb) => {
+  const campaign = await campaignsLib.findOne(campaignId, pgdb)
+
+  if (!campaign) {
+    throw new Error('Campaign not found.')
+  }
+
+  await Promise.map(grants, async fields => {
+    const grant = await pgdb.public.accessGrants.insertAndGet({
+      ...fields,
+      granterUserId: granter.id,
+      accessCampaignId: campaign.id,
+      beginBefore: addInterval(moment(), campaign.grantClaimableInterval)
+    })
+
+    await eventsLib.log(grant, 'insert', pgdb)
+  }, { concurrency: 10 })
+}
```

Comment:
looking at it now, I don't completely understand why you dont map the grants into a new array and then insert this with one `pgdb.insert`?